src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -201,9 +203,21 @@ export class OrganizationIntegrations extends AsyncComponent<
   }
 
   sortIntegrations(integrations: AppOrProviderOrPlugin[]) {
-    return integrations
-      .sort((a, b) => a.name.localeCompare(b.name))
-      .sort((a, b) => this.getInstallValue(b) - this.getInstallValue(a));
+    const sortByName = arr => arr.sort((a, b) => a.name.localeCompare(b.name));
+    const sortByPopularity = arr =>
+      arr.sort((a, b) => {
+        const weightA = weights[a.slug] ? weights[a.slug] : 1;
+        const weightB = weights[b.slug] ? weights[b.slug] : 1;
+        return weightA > weightB ? -1 : weightB > weightA ? 1 : 0;
+      });
+    const sortByInstalled = arr =>
+      arr.sort((a, b) => this.getInstallValue(b) - this.getInstallValue(a));
+
+    if (isSortIntegrationsByWeightActive()) {
+      return sortByInstalled(sortByPopularity(sortByName(integrations)));
+    }
+
+    return sortByInstalled(sortByName(integrations));
```

Comment:
This is a little difficult to read, I would probably just write out the functions for each sort:
```js
const sortByName = (a,b) => a.name.localeCompare(b.name));
const sortByInstalled = (a,b) => this.getInstallValue(b) - this.getInstallValue(a);
```
With this, you can just chain the sort methods:
```js
arr.sort(sortByName).sort(sortByInstalled)
```
But functionally, it's doing the exact same thing. Maybe get some second opinions on this I'm no expert :)

Reply:
@leeandher Good point! I like your way more.